File: base/src/main/java/com/tsy/base/tool/VersionTool.java
package com.tsy.base.tool;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

/**
 * Created by jay on 17/7/31.
 */

public final class VersionTool {

    private VersionTool() {
    }


    public static String getVersionName(Context context) {
        try {
            PackageInfo pi = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
            return pi.versionName;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
            return "3.1.0";
        }
    }

    public static String getVersionCode(Context context) {
        try {
            PackageInfo pi = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
            return pi.versionCode + "";
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
            return "0";
        }
    }



}

File: base/src/main/java/com/tsy/base/tool/VersionTool.kt
package com.tsy.base.tool

import android.content.Context
import android.content.pm.PackageManager

/**
 * Created by jay on 17/7/31.
 */

object VersionTool {


    fun getVersionName(context: Context): String {
        try {
            val pi = context.packageManager.getPackageInfo(context.packageName, 0)
            return pi.versionName
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
            return "3.1.0"
        }

    }

    fun getVersionCode(context: Context): String {
        try {
            val pi = context.packageManager.getPackageInfo(context.packageName, 0)
            return pi.versionCode.toString() + ""
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
            return "0"
        }

    }


}
